src/umpire/alloc/MallocAllocator.hpp

```diff
@@ -50,6 +58,47 @@ struct MallocAllocator {
     UMPIRE_LOG(Debug, "(ptr=" << ptr << ")");
     ::free(ptr);
   }
+
+  bool isHostPageable()
+  {
+#if defined(UMPIRE_ENABLE_HIP)
+    hipDeviceProp_t props;
+    int hdev = 0;
+    hipGetDevice(&hdev);
```

Comment:
Similar to what was done with device-specific calls in ResourceManager, can these calls also be moved down into the resource?

Reply:
Or are we saying that things in umpire/alloc are the same as things in umpire/resources?  (@davidbeckingsale , I am not sure?)

Reply:
They have to stay here.

Reply:
I think this is the best it can get. So this file is looking from the host resource memory perspective and asking if the cuda/hip platform is accessible. So this technically is down in the resource, but because i have to ask from the host resource if it can access the device platforms, I have to use the device-specific calls. If you look at the HipDeviceMemoryResource.cpp file, you'll see an example of how I ask, from the device memory resource perspective, if the host platform is accessible (so the reverse of this file, in regard to hip). 

This is some very detailed stuff, so we can have a webex or something to discuss it if you'd like.

Reply:
@davidbeckingsale, any thoughts?

Reply:
I think we should discuss this next Tuesday, but I'm happy with how this works right now. This isn't resource-specific since a camp::Platform doesn't define things like the device number, so we just have to do the best way can to determine whether or not the host memory is accessible from that platform.